### PR TITLE
rest: store reana_specification in Workflow table

### PR DESCRIPTION
**Pull request by dinosk**
* Amends test fixtures to include reana_specification field, as it is
  required for Workflow instace creation.

Depends on https://github.com/reanahub/reana-db/pull/10
Connects https://github.com/reanahub/reana-db/issues/12

Signed-off-by: Dinos Kousidis <dinos.kousidis@cern.ch>

### CODE REVIEW

File: docs/openapi.json [hunk 1]
```diff
@@ -199,27 +199,23 @@
...
                   "type": "string"
                 }
               },
               "required": [
-                "specification",
-                "type",
-                "name"
+                "reana_yaml",
+                "workflow_name",
+                "parameters"
```
Comment (tiborsimko) [inline docs/openapi.json]: Shall we take this occasion and standardise various concepts as:

- reana specification
- workflow specification
- input parameters
- operational parameters
- engine parameters
- ...

here in the schemas and everywhere?

See also https://github.com/reanahub/reana-workflow-controller/pull/124#pullrequestreview-150989990

File: reana_workflow_controller/rest.py [hunk 2]
```diff
@@ -1330,15 +1330,15 @@ def run_cwl_workflow_from_spec_endpoint(workflow):  # noqa
...
-def run_serial_workflow_from_spec(workflow, engine_parameters):
+def run_serial_workflow_from_spec(workflow, operational_parameters):
     """Run a serial workflow."""
     try:
         kwargs = {
             "workflow_uuid": str(workflow.id_),
             "workflow_workspace": workflow.get_workspace(),
-            "workflow_json": workflow.specification,
-            "workflow_parameters": workflow.parameters,
-            "engine_parameters": engine_parameters,
```
Comment (diegodelemos) [inline reana_workflow_controller/rest.py]: If this changes, a PR in each engine is needed.

Comment (dinosk) [reply reana_workflow_controller/rest.py]: The same things are passed, only the renaming from `engine_...` -> `operational_...` affects `rwe-serial`, PR is ready.

File: tests/conftest.py [hunk 3]
```diff
@@ -22,18 +22,152 @@
 from reana_workflow_controller.factory import create_app
 
 
-@pytest.fixture(scope='module')
+@pytest.fixture(scope="module")
+def tmp_shared_volume_path(tmpdir_factory):
```
Comment (diegodelemos) [inline tests/conftest.py]: If you rebase all this fixtures should go away.

**Comment by diegodelemos** [comment]
Can you bump `reana-db` version?

File: tests/conftest.py (hunk 3 above)
Comment (dinosk) [reply tests/conftest.py]: Removed, opened a PR in pytest-reana to refactor the fixtures

**Pull request merged**